src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -180,5 +182,20 @@ public static Set<ModuleTag> parseModuleTags(Collection<String> tags) throws Par
         }
         return moduleTagSet;
     }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<ModuleTag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<ModuleTag>} containing zero tags.
+     */
+    public static Optional<Set<ModuleTag>> parseModuleTagsForCommands(Collection<String> tags) throws ParseException {
+        assert tags != null;
+
+        if (tags.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> tagSet = tags.size() == 1 && tags.contains("") ? Collections.emptySet() : tags;
```

Comment:
Consider use of ```Optional.ofNullable```, it could make the code cleaner in that sense.

Reply:
To be added in testing